File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/ledger/validations/HouseholdMembershipValidator.java
package com.decrux.pocketr.api.services.ledger.validations;

import com.decrux.pocketr.api.exceptions.ForbiddenException;
import com.decrux.pocketr.api.services.household.ManageHousehold;
import java.util.UUID;
import org.springframework.stereotype.Component;

@Component
public class HouseholdMembershipValidator {

    public void validate(ManageHousehold manageHousehold, UUID householdId, long userId) {
        if (!manageHousehold.isActiveMember(householdId, userId)) {
            throw new ForbiddenException("Not an active member of this household");
        }
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/ledger/validations/HouseholdMembershipValidator.kt
package com.decrux.pocketr.api.services.ledger.validations

import com.decrux.pocketr.api.exceptions.ForbiddenException
import com.decrux.pocketr.api.services.household.ManageHousehold
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class HouseholdMembershipValidator {
    fun validate(
        manageHousehold: ManageHousehold,
        householdId: UUID,
        userId: Long,
    ) {
        if (!manageHousehold.isActiveMember(householdId, userId)) {
            throw ForbiddenException("Not an active member of this household")
        }
    }
}
